src/megatron/energon/flavors/captioning.py

```diff
@@ -22,3 +24,16 @@ class CaptioningSample(Sample):
 
 class CaptioningWebdataset(DefaultDecoderWebdatasetFactory[CaptioningSample]):
     __sample_type__ = CaptioningSample
+
+    def __init__(self, path: EPath, **kwargs):
```

Comment:
This code is repeated a lot. Can we build this into `DefaultDecoderWebdatasetFactory.__init__()` and only set a deprecation flag here?

Reply:
Well, it's to be deprecated anyways 🤷 So I thought it should be fine to just copy this text?